### PR TITLE
test: setup tests and write tests to TextInput component

### PR DESCRIPTION
**Pull request by kkirik**
- set up unit tests
- wrote tests to TextInput component

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -1,4 +1,7 @@
 {
     "extends": ["@yandex-cloud/eslint-config", "@yandex-cloud/eslint-config/prettier"],
-    "root": true
+    "root": true,
+    "env": {
+        "node": true
```
Comment (kkirik) [inline .eslintrc:5]: This is necessary in order not to include the package @yandex-cloud/eslint-config/server

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -26,3 +26,5 @@ jobs:
       run: npm run lint
     - name: Typecheck
       run: npm run typecheck
+    - name: Unit Tests
```
Comment (amje) [inline .github/workflows/ci.yml:29]: better use separate job for running tests. This job is only for files check

File: .gitignore [hunk 3]
```diff
@@ -5,3 +5,4 @@ node_modules
 /storybook-static
 /build
 /styles/**/*.css
+coverage
```
Comment (amje) [inline .gitignore:8]: `/coverage`

File: jest.config.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+module.exports = {
+    verbose: true,
+    moduleFileExtensions: ['js', 'json', 'ts', 'tsx'],
+    rootDir: '.',
+    testMatch: ['<rootDir>/src/**/*.test.ts?(x)'],
```
Comment (amje) [inline jest.config.js:5]: Is this regex or smth similar? Which files match `.ts?(x)` extension?

File: src/components/TextInput/TextInput.test.tsx [hunk 5]
```diff
@@ -0,0 +1,110 @@
+import React from 'react';
```
Comment (amje) [inline src/components/TextInput/TextInput.test.tsx:1]: Let's pull all the tests into subdirectory `__tests__`. Keep following this structure:
```
src/components/TextInput/
  __tests__/** - Test files
  __snapshots__/** - Snapshots for tests
  __stories__/** - Storybook stories
```

File: src/components/TextInput/TextInput.test.tsx [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+
+        expect(wrapper.find('[data-qa="yc-text-input__clear"]').exists()).toBeFalsy();
+    });
+
+    it('should call onChange with event function when input changes value', () => {
+        const FAKE_ON_CHANGE_FN = jest.fn();
+        const FAKE_EVENT = {target: {}} as TextInputEvent;
+
+        const wrapper = shallow(<TextInput onChange={FAKE_ON_CHANGE_FN} />);
+        wrapper.find(InputControl).props().onChange?.(FAKE_EVENT);
```
Comment (amje) [inline src/components/TextInput/TextInput.test.tsx:51]: Better not trigger function itself, but change value of the input via `simulate('change')`

File: src/components/TextInput/TextInput.test.tsx [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+        (wrapper.find(InputControl).props().controlRef as Function)(FAKE_CONTROL_REF.current);
+        wrapper.find('[data-qa="yc-text-input__clear"]').simulate('click', FAKE_EVENT);
+
+        expect(FAKE_ON_UPDATE_FN).toBeCalledWith('');
+    });
+
+    it('should pass autoComplete = on to InputControl component when component recieves autoComplete props = true', () => {
+        const wrapper = shallow(<TextInput autoComplete />);
+
+        expect(wrapper.find(InputControl).props().autoComplete).toBe('on');
```
Comment (amje) [inline src/components/TextInput/TextInput.test.tsx:102]: better check that underlying input element has correct html attribute

File: src/components/TextInput/TextInput.tsx [hunk 8]
```diff
@@ -147,9 +155,14 @@ export const TextInput = React.forwardRef<HTMLSpanElement, TextInputProps>(funct
...
 
-            {isErrorMsgVisible && <div className={b('error')}>{error}</div>}
+            {isErrorMsgVisible && (
+                <div data-qa={innerQa('error')} className={b('error')}>
+                    {error}
+                </div>
+            )}
             {hasClear && (
                 <span
+                    data-qa={innerQa('clear')}
```
Comment (amje) [inline src/components/TextInput/TextInput.tsx:165]: I'm not sure about using `data-qa` for component's internals. `data-qa` attribute was designed to be on the root element only

File: src/setupTests.ts [hunk 9]
```diff
@@ -0,0 +1,4 @@
+import {configure} from 'enzyme';
```
Comment (amje) [inline src/setupTests.ts:1]: Let's name file `setup-tests.ts`. Default naming is kebab-case

**Review by amje — CHANGES_REQUESTED** [review]

File: src/components/TextInput/TextInput.test.tsx (hunk 6 above)
Comment (kkirik) [reply src/components/TextInput/TextInput.test.tsx:51]: It is not impossible. Here is unit tests and we can't simulate event inside other components

File: src/components/TextInput/TextInput.tsx (hunk 8 above)
Comment (kkirik) [reply src/components/TextInput/TextInput.tsx:165]: ok, how can I find this element without data-qa attribute? by class?

File: src/components/TextInput/TextInput.test.tsx (hunk 7 above)
Comment (kkirik) [reply src/components/TextInput/TextInput.test.tsx:102]: our component has some logic about creation autoComplete props and we can test it too

File: jest.config.js (hunk 4 above)
Comment (kkirik) [reply jest.config.js:5]: ts, tsx

File: src/components/TextInput/TextInput.tsx (hunk 8 above)
Comment (amje) [reply src/components/TextInput/TextInput.tsx:165]: by class or by tag

File: src/components/TextInput/TextInput.test.tsx (hunk 7 above)
Comment (amje) [reply src/components/TextInput/TextInput.test.tsx]: ok

File: package.json [hunk 10]
```diff
@@ -19,7 +19,7 @@
         "styles"
     ],
     "scripts": {
-        "test": "jest",
+        "test:unit": "jest",
```
Comment (amje) [inline package.json:22]: we nust have `test` npm script in order to use our-release action

File: jest.config.js (hunk 4 above)
Comment (amje) [reply jest.config.js:5]: we can remove this line, because jest default is `[ "**/__tests__/**/*.[jt]s?(x)", "**/?(*.)+(spec|test).[jt]s?(x)" ]`

**Review by amje — CHANGES_REQUESTED** [review]

**Review by amje — APPROVED** [review]

**Pull request merged**